### DISCUSSION TITLE
Is ruby-kafka backwards compatible with older brokers?

### DISCUSSION THREAD

**Discussion by wbushey** [zendesk/ruby-kafka#878] [Q&A] [answered]
Hi everybody, I've looked through the README and the wiki, and I can't find any mention of whether a newer version of ruby-kafka can work with older Kafka brokers. I see that [ruby-kafka 1.0 supports Kafka 2.3 and 2.4](https://github.com/zendesk/ruby-kafka/blob/master/CHANGELOG.md); but does it also work with brokers running 2.2, or 1.0 or 0.11? Similarly, the [Compatibility table in the README](https://github.com/zendesk/ruby-kafka#compatibility) shows that ruby-kafka 0.7 supports Kafka 0.11; does it also work with brokers running 0.10.X or older?

**Comment by dasch** [comment, marked as answer]
Ruby-kafka is generally not compatible with older versions of Kafka. You'd have to use an older version of ruby-kafka that's compatible with the version of Kafka that you're targeting.

I believe that rdkafka-ruby and higher level libraries such as Racecar should have more flexible compatibility.
